walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,49 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.PriorityQueue;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    PriorityQueue<Event> eventQueue = new PriorityQueue<>(Event.ORDER_BY_START_TIME);
+    eventQueue.addAll(events);
+
+    Collection<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    int currStartTime = TimeRange.START_OF_DAY;
+
+    while (!eventQueue.isEmpty()) {
+      Event currEvent = eventQueue.remove();
+      int currEventStart = currEvent.getWhen().start();
+      int currEventEnd = currEvent.getWhen().end();
+
+      if (hasAttendees(currEvent, request.getAttendees())) {
+        int timeBlock = currEventStart - currStartTime;
```

Comment:
FYI: you can also use `currEvent.getWhen().duration()`.

Reply:
Oh I don't need the current event's duration. currStartTime is the first time all the meeting attendees are free, so currEventStart is the first time they're not all free. timeBlock is more like availableTimeBlock

Reply:
Ah, I see, you're right of course. Well then: better to change the variable names to make this more clear!